### PR TITLE
Bug 1282907 - Fix named models to be more sane

### PR DESCRIPTION
**Pull request by wlach**
* Remove unused members from NamedModel (define in subclass where really
  needed)
* Enforce uniqueness on name properties

---
This change is [Reviewable](https://reviewable.io/reviews/mozilla/treeherder/1624)

### CODE REVIEW

**Comment by edmorley** [comment]
Awesome!

I wonder if there are any dupes we need to remove first?

File: treeherder/model/models.py [hunk 1]
```diff
@@ -39,9 +39,7 @@
 @python_2_unicode_compatible
 class NamedModel(models.Model):
     id = models.AutoField(primary_key=True)
-    name = models.CharField(max_length=50, db_index=True)
-    description = models.TextField(blank=True)
```
Comment (edmorley) [inline treeherder/model/models.py]: Since we're removing description here, we'll need to add it back for `RepositoryGroup`.

Comment (wlach) [inline treeherder/model/models.py]: Yup, just noticed that. Fixed. :)

**Comment by wlach** [comment]
I don't think any duplicates are likely in a production context. All the named models had a minimal number of rows, half of them were generated from fixtures.

**Pull request merged**